specs/commandline/plugin.md

```diff
@@ -41,51 +43,151 @@ Aliases:
 ### notation plugin install
 
 ```text
-Installs a plugin
+Install a plugin
 
 Usage:
-  notation plugin install [flags] <plugin path>
+  notation plugin install [flags] <plugin_source>
 
 Flags:
   -h, --help                    help for install
   -f, --force                   force the installation of a plugin
-
+      --checksum string         must match SHA256 of the plugin source
+      --source string           the plugin installation source, options: "file", "url","registry" (default "file")                  
 Aliases:
   install, add
 ```
 
-### notation plugin remove
+### notation plugin upgrade
 
 ```text
-Removes a plugin
+Upgrade a plugin
 
 Usage:
-  notation plugin remove [flags] <plugin name>
+  notation plugin upgrade [flags] <plugin_source>
 
 Flags:
-  -h, --help          help for remove
+  -h, --help                    help for upgrade
+      --checksum string         must match SHA256 of the plugin source
+      --source string           the plugin installation source, options: "file", "url","registry" (default "file")                  
+```
+
+### notation plugin uninstall
+
+```text
+Uninstall a plugin
+
+Usage:
+  notation plugin uninstall [flags] <plugin_name>
 
+Flags:
+  -h, --help          help for remove
+  -y, --yes           do not prompt for confirmation
 Aliases:
-  remove, rm, uninstall, delete
+  remove, rm, uninstall
 ```
 
 ## Usage
 
-### Install a plugin
+## Install a plugin 
+
+### Install a plugin from file system
+
+Install a Notation plugin from file system and verify the plugin checksum.
+
+```shell
+$ notation plugin install --source file --checksum <digest> <file_path>
```

Comment:
When install from a local file, the checksum should be optional. Otherwise, people will just do
```bash
notation plugin install --source file --checksum $(sha256sum $file | cut -f1 -d' ') $file
```
which is not really useful.